motionMonitor/motionMonitor.c

```diff
@@ -76,41 +103,48 @@ le_result_t brnkl_motion_getSuddenImpact(double* xAcc,
   return LE_OK;
 }
 
-int8_t brnkl_motion_hasSuddenImpact() {
-  return hasSuddenImpact;
-}
+/*
+*Monitors accelerometer from iio on 100ms intervals
+*Sets hasSuddenImpact flag when accelerometer surpasses threshold
+*/
+void *impactMonitor(void * ptr){
 
-// take a reading but make sure its "larger" than
-// whatever is already in the buffer
-void interruptChangeHandler(bool state, void* ctx) {
   double x, y, z;
-  le_result_t r = brnkl_motion_getCurrentAcceleration(&x, &y, &z);
-  LE_INFO("TRIGGERED x:%f y:%f z:%f", x, y, z);
-  if (r == LE_OK) {
-    bool updateX = fabs(x) > fabs(suddenImpact.x);
-    bool updateY = fabs(y) > fabs(suddenImpact.y);
-    bool updateZ = fabs(z) > fabs(suddenImpact.z);
-    if (updateX) {
-      suddenImpact.x = x;
-    }
-    if (updateY) {
-      suddenImpact.y = y;
-    }
-    if (updateZ) {
-      suddenImpact.z = z;
-    }
-    if (updateX || updateY || updateZ) {
+
+  for(;;){
+
+    brnkl_motion_getCurrentAcceleration(&x, &y, &z);
+
+    if(
+      abs(x) + 
+      abs(y) + 
+      abs(z) > 
+      impactThreshold
+      )
       hasSuddenImpact = true;
-    }
+    
+
+    usleep(100*1000);
+
   }
+  return ptr;
 }
 
-void initGpio() {
-  interrupt_SetInput(INTERRUPT_ACTIVE_HIGH);
-  interrupt_AddChangeEventHandler(INTERRUPT_EDGE_RISING, interruptChangeHandler,
-                                  NULL, 0);
+/*
+*Create thread to monitor accelerometer iio
+*/
+void initThread(){
+  int thread;
+
+  thread = pthread_create( &impact_thread, NULL, impactMonitor, NULL);
+
+  if(thread){
```

Comment:
I had a quick look at the the `pthread_create` [manual](http://man7.org/linux/man-pages/man3/pthread_create.3.html) and it seems that it returns `0` on success, and an error code on error. I'm not very familiar with the pthread API so I may be missing something.
```suggestion
  if(!thread){ // thread == 0 works too
```